rust/cubesql/cubesql/src/compile/rewrite/rules/filters.rs

```diff
@@ -4022,6 +4064,48 @@ impl FilterRules {
                         new_end_date.format("%Y-%m-%d").to_string(),
                     ]
                 }
+                "quarter" | "qtr" => {
+                    // Check that the range only covers one year
+                    let start_date_year = start_date.year();
+                    if start_date_year != end_date.year() {
+                        return false;
+                    }
+
+                    // Quarter value must be valid (1-4)
+                    if !(1..=4).contains(&value) {
+                        return false;
+                    }
+
+                    let quarter_start_month = (value - 1) * 3 + 1;
+
+                    // Obtain the new range
+                    let Some(new_start_date) =
+                        NaiveDate::from_ymd_opt(start_date_year, quarter_start_month as u32, 1)
+                    else {
+                        return false;
+                    };
+
+                    let Some(new_end_date) = new_start_date
+                        .checked_add_months(Months::new(3))
+                        .and_then(|date| date.checked_sub_days(Days::new(1)))
+                    else {
+                        return false;
+                    };
+
+                    // Paranoid check, If the resulting range is outside of the original range, we can't merge
```

Comment:
Someone can combine timestamp comparision and quarter. Let's add these check to protect such cases.